enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -199,6 +214,16 @@ def handle_reversing_enterprise_course_unenrollment(self, unenrollment):
                 fulfillment_uuid,
                 enrollment_unenrolled_at
             )
+
+            for external_reference in related_transaction.external_reference.all():
+                provider_slug = external_reference.external_fulfillment_provider.slug
+                if provider_slug == GEAGFulfillmentHandler.EXTERNAL_FULFILLMENT_PROVIDER_SLUG:
+                    # this will raise if there is a problem before we reverse the ledger transaction
+                    # allowing us to try again later
+                    self.geag_client.cancel_enterprise_allocation(external_reference.external_reference_id)
+                else:
+                    raise FulfillmentException(f'dont know how to cancel {provider_slug}')
```

Comment:
Maybe we should move this into `GEAGFulfillmentHandler`, perhaps adding a new method to the interface like `reverse_fulfillment(transaction)`?

Reply:
i really like this and i will implement it, but i am going to incrementally merge this and circle back. i have a few balls in the air and its easier not to let perfection stand in the way of progress.